rfcs/031-monorepo.md

```diff
@@ -0,0 +1,100 @@
+- Feature Name: monorepo
+- Start Date: 2021-03-30
+- RFC PR: [amundsen-io/rfcs#0030](https://github.com/amundsen-io/rfcs/pull/30)
+- Amundsen Issue: [amundsen-io/amundsen#0000](https://github.com/amundsen-io/amundsen/issues/0000) (leave this empty for now)
+
+# Move to Monorepo
+
+## Summary
+
+At the moment, Amnundsen project is divided into 6 repositories: 
+	- amundsen
+	- amundsenfrontendlibrary
+	- amundsenmetadatalibrary
+	- amundsensearchlibrary
+	- amundsencommon 
+	- amudnsendatabuilder
+
+This RFC proposes merging all of these repositories into a single repo.
+
+There will be no change to the built artifacts. Users who use PyPi packages or Docker images will have no change to their deployments. Users with forks or submodule based customization will have to make minor changes to their CI/CD pipelines to pull from the correct directories, but the required changes will be minor.
```

Comment:
Smooth change management, nice !

However, with current structure, when someone is using the submodule approach can freely choose versions (even refs) for each product separately (I can use frontend in different version than metadata and search). This change somehow limits that capability. It's worth to explicitly mention it in drawbacks.

Reply:
ah, good point, added